### PR TITLE
Split try/catch from try/finally

### PR DESCRIPTION
**Pull request by bakkot**

### CODE REVIEW

File: src/main/java/com/shapesecurity/shift/es2016/semantics/asg/TryCatch.java [hunk 1]
```diff
@@ -23,33 +23,28 @@
 
 import java.util.Objects;
 
-public class TryCatchFinally implements Node {
+public class TryCatch implements Node {
 	@Nonnull
 	public final Block tryBody;
 	@Nonnull
-	public final Maybe<Pair<Variable, Block>> catchBody;
+	public final Pair<Variable, Block> catchBody;
```
Comment (michaelficarra) [inline src/main/java/com/shapesecurity/shift/es2016/semantics/asg/TryCatch.java:30]: Maybe we want to make the `Variable` optional?

**Review by michaelficarra — APPROVED** [review]
I <3 this PR so much. LGTM otherwise.

File: src/main/java/com/shapesecurity/shift/es2016/semantics/asg/TryCatch.java (hunk 1 above)
Comment (bakkot) [reply src/main/java/com/shapesecurity/shift/es2016/semantics/asg/TryCatch.java:30]: Eh, not until 2018 at the earliest, I think.

**Comment by bakkot** [comment]
Now includes version bump commits.

**Pull request merged**